controller_manager:
  ros__parameters:
    update_rate: 1000 # Hz

    gravity_compensation_controller:
      type: gravity_compensation_controller/GravityCompensationController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    franka_robot_state_broadcaster:
      type: franka_robot_state_broadcaster/FrankaRobotStateBroadcaster

franka_robot_state_broadcaster:
  ros__parameters:
    arm_id: panda

gravity_compensation_controller:
  ros__parameters:
    arm_id: panda
    joint_names:
      - panda_joint1
      - panda_joint2
      - panda_joint3
      - panda_joint4
      - panda_joint5
      - panda_joint6
      - panda_joint7
    kp_gains:
      - 0.1
      - 0.1
      - 0.1
      - 0.1
      - 0.1
      - 0.1
      - 0.1
    kd_gains:
      - 0.01
      - 0.01
      - 0.01
      - 0.01
      - 0.01
      - 0.01
      - 0.01
    torque_limits:
      - 2.0
      - 2.0
      - 2.0
      - 2.0
      - 2.0
      - 2.0
      - 2.0
